### ISSUE TITLE
Convert decimal type to float?

### ISSUE THREAD

**Issue by GaryJones** [phpro/soap-client#37]
The XML schemas supports a numeric data type of `decimal`. Right now, the WDSL I'm parsing has:

```xml
<s:element minOccurs="1" maxOccurs="1" name="Fee" nillable="true" type="s:decimal" />
```
Which results in a generated class property of:
```php
   /**
     * @var decimal
     */
    protected $Fee = null;
```

and a corresponding getter:

```php
    /**
     * @return decimal
     */
    public function getFee()
    {
        return $this->Fee;
    }
```

Could this be automatically converted to a float by this package, or otherwise handled in a nicer manner?

**Comment by GaryJones** [comment]
I could they could be achieved [here](https://github.com/phpro/soap-client/blob/8229e08c33288960b57f9556ac5292a4644afcb5/src/Phpro/SoapClient/CodeGenerator/Util/Normalizer.php#L48-L57).

Also, `double` could be normalized as well?

**Comment by HKandulla** [comment]
Is there someone already working on this? Otherwise, I could prepare a PR over the next few days?

**Comment by veewee** [comment]
Nobody is currently working on this. Feel free to pick it up. Thanks! :)

**Comment by HKandulla** [comment]
great:-)

**Comment by GaryJones** [comment]
This is what I created for my own project that seemed to work. Might be a starting point for the PR.

https://gist.github.com/GaryJones/5704cb3ad88fc3cdb01e255c16914d84

No claims to any of it being right :-)

**Comment by veewee** [comment]
@GaryJones: 
Oh okay, I thought the issue was about the type hinting and did not know there was also a conversion issue. I would assume that the soap extension was smart enough to convert XSD decimals to PHP floats, but in this case a type converter is the correct solution.

There is one small remark I have about your code: it should never return null. If it does not exist, then the default for a float is `0.0`.

**Comment by HKandulla** [comment]
@GaryJones Thanks for the gist. Here: https://github.com/phpro/soap-client/pull/38 is the PR. It is basically @GaryJones TypeConverter + test and the float typehint support for decimal. Shall I do the same for double and add it to the PR?

**Comment by GaryJones** [comment]
I've got a similar code for the `double`, but it's a copy / paste / amend from the Decimal code, so you may as well adapt the code in your PR if you want to add it :-)

**Comment by veewee** [comment]
Fixed in #38
